slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,378 @@
+---
+title: "Extras - Recipes"
+subtitle: "Introduction to tidymodels"
+format:
+  revealjs: 
+    slide-number: true
+    footer: <https://workshops.tidymodels.org>
+    include-before-body: header.html
+    include-after-body: footer-annotations.html
+    theme: [default, tidymodels.scss]
+    width: 1280
+    height: 720
+knitr:
+  opts_chunk: 
+    echo: true
+    collapse: true
+    comment: "#>"
+    fig.path: "figures/"
+---
+
+```{r}
+#| label: setup
+#| include: false
+#| file: setup.R
+```
+
+## Looking at the predictors
+
+```{r}
+#| label: setup-previous
+#| echo: false
+library(countdown)
+library(tidymodels)
+library(modeldatatoo)
+taxi <- data_taxi()
+
+set.seed(123)
+taxi_split <- initial_split(taxi, prop = 0.8, strata = tip)
+taxi_train <- training(taxi_split)
+taxi_test <- testing(taxi_split)
+
+tree_spec <- decision_tree(cost_complexity = 0.0001, mode = "classification")
+taxi_wflow <- workflow(tip ~ ., tree_spec)
+taxi_fit <- fit(taxi_wflow, taxi_train)
+```
+
+```{r}
+#| label: more-setup
+#| include: false
+library(countdown)
+
+options(width = 200)
+
+ggplot2::theme_set(ggplot2::theme_bw())
+```
+
+```{r}
+#| label: taxi_train
+taxi_train
+```
+
+## Working with other models
+
+Some models can't handle non-numeric data
+
+-   Linear Regression
+-   K Nearest Neighbors
+
+<br>
+
+::: fragment
+Some models struggle if numeric predictors aren't scaled
+
+-   K Nearest Neighbors
+-   Anything using gradient descent
+:::
+
+## Types of needed preprocessing
+
+-   Do qualitative predictors require a numeric encoding?
+
+-   Should columns with a single unique value be removed?
+
+-   Does the model struggle with missing data?
+
+-   Does the model struggle with correlated predictors?
+
+-   Should predictors be centered and scaled?
+
+-   Is it helpful to transform predictors to be more symmetric?
+
+::: footer
+<https://www.tmwr.org/pre-proc-table.html>
+:::
+
+## Two types of preprocessing
+
+![](images/fe_venn.svg){fig-align="center"}
```

Comment:
I'm super into this graphic.🥳

Thoughts on labeling the second type "Transformation, normailization, ..." as data preprocessing in the graphic to tighten the relationship between the graphic and the following slide?

Reply:
That would be a good idea, but I don't have the source for this graphic, as it was already in the repo 😄